tests/template_tests/test_nodelist.py

```diff
@@ -73,3 +73,24 @@ def test_correct_exception_index(self):
             except (RuntimeError, TypeError) as e:
                 debug = e.template_debug
                 self.assertEqual((debug['start'], debug['end']), expected_error_source_index)
+
+    def test_stream_correct_exception_index(self):
+        tests = [
+            ('{% load bad_tag %}{% for i in range %}{% badsimpletag %}{% endfor %}', (38, 56)),
+            ('{% load bad_tag %}{% for i in range %}{% for j in range %}{% badsimpletag %}{% endfor %}{% endfor %}', (58, 76)),
+            ('{% load bad_tag %}{% for i in range %}{% badsimpletag %}{% for j in range %}Hello{% endfor %}{% endfor %}', (38, 56)),
+            ('{% load bad_tag %}{% for i in range %}{% for j in five %}{% badsimpletag %}{% endfor %}{% endfor %}', (38, 57)),
+            ('{% load bad_tag %}{% for j in five %}{% badsimpletag %}{% endfor %}', (18, 37)),
+        ]
+        context = Context({
+            'range': range(5),
+            'five': 5,
+        })
+        engine = Engine(debug=True, libraries={'bad_tag': 'template_tests.templatetags.bad_tag'})
+        for source, expected_error_source_index in tests:
+            template = engine.from_string(source)
+            try:
+                ''.join(template.stream(context))
+            except (RuntimeError, TypeError) as e:
+                debug = e.template_debug
+                self.assertEqual((debug['start'], debug['end']), expected_error_source_index)
```

Comment:
Since we're adding a new code path where ``template_debug`` information is added to the exception, my main concern is that the path is exercised in the tests.

Some duplication could be avoided here by removing everything except lines 92-96 and inserting that in the same loop that tests ``template.render``. It seems okay to me here to add the new checks in the same test case.